File: examples/pinecone/python/quickstart.py
"""
Embex Quick Start - Pinecone Provider

Requires: Pinecone API key and index name
Set: export PINECONE_API_KEY="your-api-key"
Run: python examples/pinecone/python/quickstart.py
"""
import asyncio
import os
from embex import EmbexClient, Point


async def main():
    print("🚀 Embex Quick Start - Pinecone Provider\n")
    
    # Get API key from environment
    api_key = os.getenv("PINECONE_API_KEY")
    if not api_key:
        print("❌ Error: PINECONE_API_KEY environment variable not set")
        print("   Set it with: export PINECONE_API_KEY='your-api-key'")
        return
    
    # Pinecone - requires API key
    # URL format: https://<index-name>-<project-id>.svc.<environment>.pinecone.io
    # Or use: https://api.pinecone.io for serverless
    index_name = os.getenv("PINECONE_INDEX_NAME", "embex-quickstart")
    url = f"https://{index_name}.svc.pinecone.io"  # Adjust based on your Pinecone setup
    
    print(f"📋 Connecting to Pinecone index: {index_name}\n")
    
    client = EmbexClient(provider="pinecone", url=url, api_key=api_key)
    
    collection_name = "documents"
    collection = client.collection(collection_name)
    
    # Clean up if exists
    try:
        await collection.delete_collection()
        print(f"✅ Cleaned up existing collection: {collection_name}")
    except Exception:
        pass
    
    # 1. Create Collection
    print(f"\n📦 Creating collection: {collection_name}")
    await collection.create(dimension=768, distance="cosine")
    print("   ✅ Collection created!")
    
    # 2. Insert Data
    print("\n📝 Inserting documents...")
    points = [
        Point(id="1", vector=[0.1] * 768, metadata={"title": "Hello World"}),
        Point(id="2", vector=[0.2] * 768, metadata={"title": "Embex is Fast"}),
        Point(id="3", vector=[0.15] * 768, metadata={"title": "Rust Powered"}),
    ]
    await collection.insert(points)
    print(f"   ✅ Inserted {len(points)} documents!")
    
    # 3. Search
    print("\n🔍 Searching...")
    results = await collection.search(vector=[0.12] * 768, top_k=2)
    print(f"   ✅ Found {len(results.results)} results:\n")
    for i, result in enumerate(results.results, 1):
        print(f"   {i}. {result.metadata.get('title', 'N/A')} (Score: {result.score:.4f})")
    
    print("\n🎉 Pinecone quick start complete!")
    print("\n💡 Note: Pinecone is serverless - no local setup needed!")


if __name__ == "__main__":
    asyncio.run(main())

File: examples/pinecone/node/quickstart.ts
/**
 * Embex Quick Start - Pinecone Provider
 *
 * Requires: Pinecone API key and index name
 * Set: export PINECONE_API_KEY="your-api-key"
 * Run: npx tsx examples/pinecone/node/quickstart.ts
 */

import { EmbexClient } from "@bridgerust/embex";

async function main() {
  console.log("🚀 Embex Quick Start - Pinecone Provider\n");

  // Get API key from environment
  const apiKey = process.env.PINECONE_API_KEY;
  if (!apiKey) {
    console.error("❌ Error: PINECONE_API_KEY environment variable not set");
    console.error("   Set it with: export PINECONE_API_KEY='your-api-key'");
    process.exit(1);
  }

  // Pinecone - requires API key
  // URL format: https://<index-name>-<project-id>.svc.<environment>.pinecone.io
  // Or use: https://api.pinecone.io for serverless
  const indexName = process.env.PINECONE_INDEX_NAME || "embex-quickstart";
  const url = `https://${indexName}.svc.pinecone.io`; // Adjust based on your Pinecone setup

  console.log(`📋 Connecting to Pinecone index: ${indexName}\n`);

  const client = new EmbexClient("pinecone", url, apiKey);

  const collectionName = "documents";
  const collection = client.collection(collectionName);

  // Clean up if exists
  try {
    await collection.deleteCollection();
    console.log(`✅ Cleaned up existing collection: ${collectionName}`);
  } catch (e) {
    // Collection doesn't exist
  }

  // 1. Create Collection
  console.log(`\n📦 Creating collection: ${collectionName}`);
  await collection.create(768, "cosine");
  console.log("   ✅ Collection created!");

  // 2. Insert Data
  console.log("\n📝 Inserting documents...");
  const points = [
    {
      id: "1",
      vector: Array(768).fill(0.1),
      metadata: { title: "Hello World" },
    },
    {
      id: "2",
      vector: Array(768).fill(0.2),
      metadata: { title: "Embex is Fast" },
    },
    {
      id: "3",
      vector: Array(768).fill(0.15),
      metadata: { title: "Rust Powered" },
    },
  ];
  await collection.insert(points);
  console.log(`   ✅ Inserted ${points.length} documents!`);

  // 3. Search
  console.log("\n🔍 Searching...");
  const results = await collection.search(Array(768).fill(0.12), 2);
  console.log(`   ✅ Found ${results.results.length} results:\n`);
  results.results.forEach((result, i) => {
    console.log(
      `   ${i + 1}. ${
        result.metadata?.title || "N/A"
      } (Score: ${result.score.toFixed(4)})`
    );
  });

  console.log("\n🎉 Pinecone quick start complete!");
  console.log("\n💡 Note: Pinecone is serverless - no local setup needed!");
}

main().catch(console.error);
